client/client.go

```diff
@@ -362,6 +383,11 @@ func (c *CfClient) getSegmentsFromCache(fc *evaluation.FeatureConfig) {
 	}
 }
 
+func (c *CfClient) setAnalyticsServiceClient(ctx context.Context) {
+	<-c.authenticated
+	c.analyticsService.StartClient(ctx, &c.metricsapi, c.environmentID)
```

Comment:
After we authenticate we should send a target event (this includes the target data and is used as part of target auto-discovery).  It might fit in nicely here after you start the service.

Also I wonder if StartClient should just be Start ?